[ADD] Filter by groups on default view get

corsojulian7 opened the pull request:
Description of the issue/feature this PR addresses: It allows you to create custom views for each user group.

Current behavior before PR: Groups are ignored when odoo is searching for the default view.

Desired behavior after PR is merged: Search filtering by userg groups




--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

mart-e commented:
Hello,

I am not convinced about this.
Creating views is something technical, it is not really a problem to create a view if the current user does not belong in the group (e.g. creating views for logistic even if not in logistic).

Also, it seems you have not [signed the CLA](https://www.odoo.com/sign-cla).
Could you add it?
Regards

corsojulian7 commented:
> Hello,
> 
> I am not convinced about this.
> Creating views is something technical, it is not really a problem to create a view if the current user does not belong in the group (e.g. creating views for logistic even if not in logistic).
> 
> Also, it seems you have not [signed the CLA](https://www.odoo.com/sign-cla).
> Could you add it?
> Regards

Hello. This has nothing to do with creating views. It is useful for odoo to search for existing views ignoring those that do not belong to the user's group, if it is assigned. So you can make views with more priority than the native one, but assigning a restrictive group. In this case the user in these groups will access the custom view while the user without it will access the original view.
For example, if in the native view of sale_order we assign the group "sales manager" and create a lower priority that is an empty form, the base_user will access a blank page while the managers will see all the information. It is a useless example but it serves to imagine the administration of permissions that this allows.
Regards!

mart-e commented:
Ok, thanks for the update, I understand better now.
The issue is that the `groups_id` is taken into account when computing the inheritance of the views but not when selection the "base" view.
As the "Groups" is used in inheritance, I can see the logic to apply it as well in the base view selection.
However, this should definitely target the master branch as it may break some systems with incorrect "Groups" value.

@odony do you have any objection for this change or see other potential problems?

corsojulian7 commented:
Hi @odony, @mart-e ! My merge was approved, and passed ci/runbot test.
I updated the signature, and now i can't pass the ci/runbot test.
I changed anything.

xmo-odoo commented:
> Hi @odony , @xmo-odoo , @mart-e ! My merge was approved, and passed ci/runbot test.
> I updated the signature, and now i can't pass the ci/runbot test.
> I changed anything.

Yeah don't worry, we moved the builder/runner to https and it looks like some bits of the infrastructure (e.g. the headless chrome running tests) are... not enjoying the experience. I pinged @d-fence to try and take a look at it as he's the RunBoss.

d-fence commented:
@corsojulian7 the [build](https://runbot.odoo.com/runbot/build/690205) that passed the ci/runbot was ran on 2019-11-15. Since that, the chrome headless instrumentation changed.
I think that you should rebase your branch and it should go green again.

corsojulian7 commented:
@d-fence Thank you, now it passed all tests.

corsojulian7 commented:
Hello @odony , @xmo-odoo , @mart-e. How can I merge the changes? Changes are approved and all checks have passed.
Thanks!

mart-e commented:
@corsojulian7 my previous comment still stands

>  However, this should definitely target the master branch as it may break some systems with incorrect "Groups" value.

Could you retarget?

corsojulian7 commented:
@mart-e I changed the branch, and it throwed me a lot of conflicts. How can I continue?

mart-e commented:
@corsojulian7 the easiest for a single patch is to create a new branch based on the master. If you want to keep the same, you have to do something like
```
$ git reset odoo/master --hard
$ git cherry-pick <hash of the commits you want>
$ git push -f
```

corsojulian7 commented:
@mart-e I changed the branch to master, and now all checks have passed. What i need to do now?

mart-e commented:
Hi Julian,

Sorry we are still discussing internally about this change. We are afraid this could break some code, for instance if no view is available due to the group attribute, the user would endup with the basic autogenerated view, not great...

corsojulian7 commented:
@mart-e This shouldn't break nothing, because every model must have a view to display if the user is not in a group (by default odoo has it). The "problem" is when the odoo's administrator adds a group to a default view. But that is a user mistake.
